sdks/python/apache_beam/io/source_test_utils.py

```diff
@@ -67,6 +72,40 @@ class ExpectedSplitOutcome(object):
   MUST_BE_CONSISTENT_IF_SUCCEEDS = 3
 
 
+class TestCaseWithTempDirCleanUp(unittest.TestCase):
+  """Base class for TestCases that deals with TempDir clean-up.
+
+  Inherited test cases will call self._new_tempdir() to start a temporary dir
+  which will be deleted at the end of the tests (when tearDown() is called).
+  """
+
+  def setUp(self):
+    self._tempdirs = []
+
+  def tearDown(self):
+    for path in self._tempdirs:
```

Comment:
Can this class hold just 1 temp directory rather than multiple?

Reply:
Changed to context manager.